### PR TITLE
tiflash: add 6.2.0 upgrade guide

### PR DESCRIPTION
**Pull request by shichun-0415**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/blob/master/tiflash-620-upgrade-guide.md, which involves the following PRs:
    - https://github.com/pingcap/docs-cn/pull/9615
    - https://github.com/pingcap/docs-cn/pull/10763
    - https://github.com/pingcap/docs-cn/pull/10866
    - https://github.com/pingcap/docs-cn/pull/10938
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tiflash-610-upgrade.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+- [Upgrade TiDB Using TiUP](/upgrade-tidb-using-tiup.md)
+- [Upgrade TiDB in Kubernetes](https://docs.pingcap.com/tidb-in-kubernetes/stable/upgrade-a-tidb-cluster)
+
+## Upgrade policies
+
+It is not recommended that you upgrade TiFlash to v6.1.0 across major versions, for example, from v4.0.0 to v6.1.0. Instead, you need to upgrade earlier versions to v5.4.x or v6.0.0 first, and then to v6.1.0.
+
+### Upgrade from v4.x.x to v5.x.x
+
+v4.x.x is near the end of its life cycle. Please upgrade to v5.x.x or later as soon as possible.
```
Comment (qiancai) [inline tiflash-610-upgrade.md:21]:
```suggestion
v4.x.x is near the end of its life cycle. It is recommended that you upgrade to v5.x.x or later as soon as possible.
```

File: tiflash-610-upgrade.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+You can forcibly scale in the target TiFlash node and then replicate data. For detailed steps, see [Scale in a TiFlash cluster](/scale-tidb-using-tiup.md#scale-in-a-tiflash-cluster).
+
+#### Dynamic pruning
+
+If you do not enable dynamic pruning and will not use it in the future, you can skip this section.
+
+- Newly installed TiDB v6.1.0: Dynamic pruning is enabled by default.
+
+- TiDB v6.0.0 and earlier: Dynamic pruning is disabled by default. The setting of dynamic pruning inherits the previous version after an upgrade. That is, dynamic pruning will not be enabled (or disabled) automatically after an upgrade.
```
Comment (qiancai) [inline tiflash-610-upgrade.md:45]:
```suggestion
- TiDB v6.0.0 and earlier: Dynamic pruning is disabled by default. The setting of dynamic pruning after an upgrade inherits that of the previous version. That is, dynamic pruning will not be enabled (or disabled) automatically after an upgrade.
```

File: tiflash-610-upgrade.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+    After an upgrade, to enable dynamic pruning, you need to manually update GlobalStats of partitioned tables. For details, see [Dynamic pruning mode](/partitioned-table.md#dynamic-pruning-mode).
+
+#### TiFlash PageStorage
+
+By default, TiFlash v6.1.0 upgrades PageStorage to the V3 version (`format_version=4`). This version significantly reduces the peak write I/O traffic and CPU usage caused by TiFlash data GC in scenarios with high concurrency and heavy queries.
```
Comment (qiancai) [inline tiflash-610-upgrade.md:51]:
```suggestion
By default, TiFlash v6.1.0 upgrades PageStorage to the V3 version (corresponding to `format_version=4` in the configuration file). This version significantly reduces the peak write I/O traffic and CPU usage caused by TiFlash data GC in scenarios with high concurrency or heavy queries.
```

File: tiflash-610-upgrade.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+- With more data written to the existing TiFlash nodes following the upgrade to v6.1.0, earlier data will be gradually converted to the new format.
```
Comment (qiancai) [inline tiflash-610-upgrade.md:53]:
```suggestion
- With more data written to the existing TiFlash nodes after an upgrade to v6.1.0, earlier data will be gradually converted to the new format.
```

File: tiflash-610-upgrade.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+- However, earlier data cannot be completely converted to the new format, because the conversion consumes certain amount of system overhead (services are not affected, but you still need to pay attention). After the upgrade, it is recommended that you run the [compaction](/sql-statements/sql-statement-alter-table-compact.md) command to convert the data to the new format. The steps are as follows:
+
+    1. Run the following command to each table containing TiFlash replicas:
```
Comment (qiancai) [inline tiflash-610-upgrade.md:56]:
```suggestion
    1. Run the following command for each table containing TiFlash replicas:
```

File: tiflash-610-upgrade.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+As a non-LTS version, v6.0.0 will not release subsequent bug-fix versions. Please use v6.1.0 and later LTS versions whenever possible.
+
+### Upgrade from 5.x.x to v6.1.0
+
+#### TiFlash Proxy
+
+TiFlash Proxy is upgraded in v6.1.0 (aligned with TiKV v6.0.0). The new version has upgraded the RocksDB version. After you upgrade TiFlash to v6.1.0, the data format is converted to the new version automatically.
+
+In normal upgrades, the data conversion does not involve any risks. However, if you need to downgrade TiFlash from v6.1.0 to any earlier version in special scenarios (for example, testing or verification scenarios), note that the RocksDB configuration of the newer version might fail to be parsed. As as result, TiFlash will fail to restart. It is recommended that you fully test and verify the upgrade process and prepare an emergency plan.
```
Comment (qiancai) [inline tiflash-610-upgrade.md:33]:
```suggestion
In regular upgrades, the data conversion does not involve any risks. However, if you need to downgrade TiFlash from v6.1.0 to any earlier version in special scenarios (for example, testing or verification scenarios), note that the RocksDB configuration of the newer version might fail to be parsed. As as result, TiFlash will fail to restart. It is recommended that you fully test and verify the upgrade process and prepare an emergency plan.
```

File: tiflash-620-upgrade-guide.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+>
+> - v4.x.x is near the end of its life cycle. Please upgrade to v5.x.x or later as soon as possible. For more information, see [TiDB Release Support Policy](https://en.pingcap.com/tidb-release-support-policy/).
+>
+> - As a non-LTS version, v6.0.0 will not release subsequent bug-fix versions. Please use v6.1.0 and later LTS versions whenever possible.
+>
+> - To upgrade TiFlash from versions earlier than V5.3.0 to V5.3.0 or later, you should stop TiFlash and then upgrade it. The following steps help you upgrade TiFlash without interrupting other components:
+> 
+>    - Stop the TiFlash instance: `tiup cluster stop <cluster-name> -R tiflash`
+>    - Upgrade the TiDB cluster without restarting it (only updating the files): `tiup cluster upgrade <cluster-name> <version> --offline`
+>    - Reload the TiDB cluster: `tiup cluster reload <cluster-name>`. After the reload, the TiFlash instance is started and you do not need to manually start it.
```
Comment (shichun-0415) [inline tiflash-620-upgrade-guide.md:27]:
```suggestion
>     - Stop the TiFlash instance: `tiup cluster stop <cluster-name> -R tiflash`
>     - Upgrade the TiDB cluster without restarting it (only updating the files): `tiup cluster upgrade <cluster-name> <version> --offline`
>     - Reload the TiDB cluster: `tiup cluster reload <cluster-name>`. After the reload, the TiFlash instance is started and you do not need to manually start it.
```

File: tiflash-620-upgrade-guide.md [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+    After an upgrade, to enable dynamic pruning, set `tidb_partition_prune_mode` to `dynamic` and manually update GlobalStats of partitioned tables. For details, see [Dynamic pruning mode](/partitioned-table.md#dynamic-pruning-mode).
+
+## From v5.x.x or v6.0.0 to v6.2.0
+
+In TiDB v6.2.0, TiFlash upgrades its data storage format to the V3 version. Therefore, when you upgrade TiFlash from v5.x.x or v6.0.0 to v6.2.0, besides functional changes in [TiFlash Proxy](#tiflash-proxy) and [Dynamic pruning](#dynamic-pruning), you also need to pay attention to the functional change in PageStorage.
+
+### PageStorage
+
+By default, TiFlash v6.2.0 uses PageStorage V3 version [`format_version = 4`](/tiflash/tiflash-configuration.md#configure-the-tiflashtoml-file). This version significantly reduces the peak write I/O traffic and relieves excessive CPU usage caused by TiFlash data GC in scenarios with high update traffic and high concurrency or heavy queries.
```
Comment (ran-huang) [inline tiflash-620-upgrade-guide.md:59]:
```suggestion
By default, TiFlash v6.2.0 uses PageStorage V3 version [`format_version = 4`](/tiflash/tiflash-configuration.md#configure-the-tiflashtoml-file). This new data format significantly reduces the peak write I/O traffic. In scenarios with high update traffic and high concurrency or heavy queries, it effectively relieves excessive CPU usage caused by TiFlash data GC.
```

File: tiflash-620-upgrade-guide.md [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+
+- With more data written to the existing TiFlash nodes after an upgrade to v6.2.0, earlier data will be gradually converted to the new format.
```
Comment (ran-huang) [inline tiflash-620-upgrade-guide.md:61]:
```suggestion
- After an upgrade to v6.2.0, as new data is written to the existing TiFlash nodes, earlier data will be gradually converted to the new format.
```

File: tiflash-620-upgrade-guide.md [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+- However, earlier data cannot be completely converted to the new format during the upgrade, because the conversion consumes a certain amount of system overhead (services are not affected, but you still need to pay attention). After the upgrade, it is recommended that you run the [`Compact` command](/sql-statements/sql-statement-alter-table-compact.md) to convert the data to the new format. The steps are as follows:
+
+    1. Run the following command for each table containing TiFlash replicas:
+
+        ```sql
+        ALTER TABLE <table_name> compact tiflash replica;
```
Comment (ran-huang) [inline tiflash-620-upgrade-guide.md:67]:
```suggestion
        ALTER TABLE <table_name> COMPACT tiflash replica;
```

File: tiflash-620-upgrade-guide.md [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+
+You can check whether there are tables still using the old data format on Grafana: Tiflash summary > storage pool > Storage Pool Run Mode.
+
+- Only V2: Number of tables using PageStorage V2 (including partitions)
+- Only V3: Number of tables using PageStorage V3 (including partitions)
+- Mix Mode: Number of tables with data format converted from PageStorage V2 to PageStorage V3 (including partitions)
+
+**Workaround for downgrading TiFlash in testing or other special scenarios**
+
+You can forcibly scale in the target TiFlash node and then replicate data. For detailed steps, see [Scale in a TiFlash cluster](/scale-tidb-using-tiup.md#scale-in-a-tiflash-cluster).
```
Comment (ran-huang) [inline tiflash-620-upgrade-guide.md:80]:
```suggestion
You can forcibly scale in the target TiFlash node and then replicate data from TiKV again. For detailed steps, see [Scale in a TiFlash cluster](/scale-tidb-using-tiup.md#scale-in-a-tiflash-cluster).
```

File: tiflash-620-upgrade-guide.md [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+
+## From v6.1.0 to v6.2.0
```
Comment (ran-huang) [inline tiflash-620-upgrade-guide.md:82]:
```suggestion
## From v6.1 to v6.2
```

**Review by hfxsd — APPROVED** [review]
LGTM

**Review by hfxsd — APPROVED** [review]
LGTM

File: tiflash-620-upgrade-guide.md [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+To learn the standard upgrade process, see the following documents:
+
+- [Upgrade TiDB Using TiUP](/upgrade-tidb-using-tiup.md)
+- [Upgrade TiDB in Kubernetes](https://docs.pingcap.com/tidb-in-kubernetes/stable/upgrade-a-tidb-cluster)
+
+> **Note:**
+>
+> - An experimental feature introduced in v6.2.0, [FastScan](/develop/dev-guide-use-fastscan.md) provides more efficient query performance at the cost of strong data consistency. Note that the form and usage of this feature might change in subsequent versions.
+>
+> - It is not recommended that you upgrade TiFlash across major versions, for example, from v4.x to v6.x. Instead, you need to upgrade from v4.x to v5.x first, and then to v6.x.
```
Comment (shichun-0415) [inline tiflash-620-upgrade-guide.md:19]:
```suggestion
> - It is not recommended that you upgrade TiDB that includes TiFlash across major versions, for example, from v4.x to v6.x. Instead, you need to upgrade from v4.x to v5.x first, and then to v6.x.
```

File: tiflash-620-upgrade-guide.md [hunk 14]
```diff
@@ -0,0 +1,86 @@
...
+
+    1. Run the following command for each table containing TiFlash replicas:
+
+        ```sql
+        ALTER TABLE <table_name> COMPACT tiflash replica;
+        ```
+
+    2. Restart the TiFlash node.
+
+You can check whether there are tables still using the old data format on Grafana: Tiflash summary > storage pool > Storage Pool Run Mode.
```
Comment (ran-huang) [inline tiflash-620-upgrade-guide.md:74]:
```suggestion
You can check whether tables still use the old data format on Grafana: **TiFlash-Summary** > **Storage Pool** > **Storage Pool Run Mode**.
```

TiFlash-Summary is used in https://docs.pingcap.com/zh/tidb/stable/monitor-tiflash

**Review by ran-huang — APPROVED** [review]
Rest LGTM

**Pull request merged**